pkg/k8splan/watcher.go

```diff
@@ -221,6 +307,27 @@ func (w *watcher) start(ctx context.Context) {
 	}
 }
 
+func (w *watcher) updateSecret(core corecontrollers.Interface, secret *v1.Secret) (*v1.Secret, error) {
+	var resultingSecret *v1.Secret
+	err := retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			if apierrors.IsConflict(err) {
+				return false
+			}
+			return true
+		},
+		func() error {
+			var err error
+			resultingSecret, err = core.Secret().Update(secret)
+			return err
+		})
+	if err == nil {
+		logrus.Debugf("[K8s] updating lastAppliedResourceVersion to %s", resultingSecret.ResourceVersion)
+		w.lastAppliedResourceVersion = resultingSecret.ResourceVersion
+	}
+	return resultingSecret, err
```

Comment:
Is there value in returning this rather than the normal `nil` value?

Reply:
I'm assuming you're referring to the `resultingSecret` -- we need to return the `resultingSecret` back to the controller framework, which is why the logic is all wonky here.